lib/menubar/utils.lua

```diff
@@ -261,6 +261,7 @@ end
 function utils.parse_desktop_file(file)
     local program = { show = true, file = file }
     local desktop_entry = false
+    local locale = string.sub(os.setlocale(nil), 1, 2) or "en"
```

Comment:
I think we should follow the spec here:
https://standards.freedesktop.org/desktop-entry-spec/latest/ar01s04.html
https://standards.freedesktop.org/desktop-entry-spec/latest/ar01s05.html

This means that we cannot use lua to query the locale since that doesn't allow to query `LC_MESSAGES`, it only has the following: "all", "collate", "ctype", "monetary", "numeric", or "time".
Also, the matching rules are more complicated than what you use here and I'm not quite sure if `lang` really is limited to two characters.

Plus, this does not only apply to `Name`, but to
- Name
- GenericName
- Comment
- Icon
- Keywords

Currently we only use Name and Icon, but I think the code should still read desktop files correctly.

Given the above complications, why do we parse `desktop` files ourselves anyway? I think it's time to switch to https://developer.gnome.org/glib/stable/glib-Key-value-file-parser.html which has `g_key_file_get_locale_string` that Does The Right Thing (tm) and is documented as implementing the Desktop Entry Specification.

What do others think about this?

Reply:
> What do others think about this?

I guess I made my opinion rather explicit about this topic in other issues: yes. And also use the icon loader if gtk is available, but that's not related to this issue.

Reply:
Using the proposed established library routine to properly extract information from the `.desktop` file will make code maintenance for this project simpler and potentially provide benefit in the future, should the sped for that file format change, and a corresponding adjustment to the library doesn't require changes to its consumers.

There's no point to proceeding with this patch if it's fated to be removed entirely anyway.